### ISSUE TITLE
For CSS EDs, auto-generate "This version" link to dated URL

### ISSUE THREAD

**Issue by SimonSapin** [tabatkins/bikeshed#523]
… dated URL like https://drafts.csswg.org/date/2015-10-25T20:04:10/css-variables/

See https://lists.w3.org/Archives/Public/www-style/2015Nov/0165.html

**Comment by tabatkins** [comment]
Hmmmmmmm. I can't just auto-gen the URL with the current time at time of generation, because that'll be *before* the draft is committed.  I could use some JS to generate the link at view-time; I'd have to observe the URL it's loaded from to see if it's already a dated version (so I could reuse the URL rather than generating a fresh incorrect URL).

Would that be ok?

**Comment by tabatkins** [comment]
Oh! Best would be to do this in Shepherd - when generating the archived docs, pass a custom metadata with the desired url.  @plinss ?

**Comment by SimonSapin** [comment]
The date is that of the commit, right? If you’re willing to integrate with hg (and maybe git) we could use the date of the current commit if the source file is "clean" (unchanged since that commit), which should always be the case for server-side bikeshedding. If it’s not, warn and don’t generate the link.

How does this sound?

**Comment by plinss** [comment]
Yeah, a custom URL during generation would be best, what do you want to call it?

**Comment by tabatkins** [comment]
"This Version", apparently.  Pass in like `bikeshed --md-\!This-Version="whatever" spec` (assuming your calling convention has problems with `!` like shell does; otherwise don't escape it).
